kernel/kernel-api/src/main/java/io/delta/kernel/internal/util/PartitionUtils.java

```diff
@@ -211,19 +218,20 @@ private static boolean hasNonPartitionColumns(
         return false;
     }
 
-    private static Predicate combineWithAndOp(Predicate left, Predicate right) {
-        String leftName = left.getName().toUpperCase();
-        String rightName = right.getName().toUpperCase();
-        if (leftName.equals("ALWAYS_FALSE") || rightName.equals("ALWAYS_FALSE")) {
-            return ALWAYS_FALSE;
-        }
-        if (leftName.equals("ALWAYS_TRUE")) {
-            return right;
-        }
-        if (rightName.equals("ALWAYS_TRUE")) {
-            return left;
+    /**
+     * Utility method to check whether the given predicate has any expressions that are not
+     * supported by the given expression handler.
+     */
+    private static boolean hasUnsupportedExpr(
+        ExpressionHandler exprHandler,
+        StructType tableSchema,
+        Predicate predicate) {
+        try {
+            exprHandler.getPredicateEvaluator(tableSchema, predicate);
+            return false;
+        } catch (UnsupportedOperationException ex) {
```

Comment:
wondering if we should throw a defined exception like `UnsupportedExpressionException`?

Reply:
Lets leave it as is for now. I think its better to think through all different exceptions and take a call.

Reply:
I agree, also worried about the cost of creating an evaluator in some engines. Is this API ok?

```
    /**
     * Is the given expression evaluation supported on the data with given schema?
     *
     * @param inputSchema Schema of input data on which the expression is evaluated.
     * @param expression Expression to evaluate.
     * @param outputType Expected result data type.
     * @return true if supported and false otherwise.
     */
    boolean isSupported(StructType inputSchema, Expression expression, DataType outputType);
```

Reply:
lets make that a separate PR. I think we need to think about whether we want the schema as well.

Reply:
Sure, I can make a separate PR. I think we need it because the connector may support an expression but it may only support it on a selected input data types.

Reply:
@tdas here is the PR: https://github.com/delta-io/delta/pull/2492